### PR TITLE
add onednn and openvino ml-benchmarks

### PR DESCRIPTION
**Pull request by dborquez**

### CODE REVIEW

File: tests/metrics/machine_learning/onednn-dockerfile/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+
+RUN apt-get update && \
+	apt-get install -y --no-install-recommends git curl build-essential autoconf && \
+	apt-get install -y --no-install-recommends php libapache2-mod-php php-bz2 php-gd php-sqlite3 php-xml && \
+	apt-get install -y --no-install-recommends cmake cmake-data mesa-utils vulkan-tools unzip apt-file wget && \
+        curl -OkL ${PHORONIX_TAR_URL}/${PHORONIX_VER} && \
+	tar zxf ${PHORONIX_VER} && cd phoronix-test-suite && ./install-sh && \
+	apt-get clean && rm -rf /var/lib/apt/lists && \
+	rm -rf /var/lib/apt/lists/* && \
```
Comment (GabyCT) [inline tests/metrics/machine_learning/onednn-dockerfile/Dockerfile:25]: is this necessary? the previous step is removing the directory

File: tests/metrics/machine_learning/onednn.sh [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+update_kata_config() {
+	readonly DEST_CONFIG="${TMP_DIR}/config_onednn.toml"
+
+	local NUM_CPUS="$(nproc --all)"
+	local MEM_AVAIL_KB="$(grep MemAvailable /proc/meminfo | awk '{print $2}')"
+	local MEM_AVAIL_MB=$(echo "scale=0; $MEM_AVAIL_KB / 1024" | bc)
+
+	info "Updating kata configuration to increase memory and cpu resources assinged to the wkld."
```
Comment (GabyCT) [inline tests/metrics/machine_learning/onednn.sh:59]: s/assinged/assigned

File: tests/metrics/machine_learning/onednn-dockerfile/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ENV PHORONIX_VER="phoronix-test-suite-10.8.4.tar.gz"
+ENV PHORONIX_CFG="/usr/share/phoronix-test-suite/pts-core/static/user-config-defaults.xml"
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+
+RUN apt-get update && \
+	apt-get install -y --no-install-recommends git curl build-essential autoconf && \
+	apt-get install -y --no-install-recommends php libapache2-mod-php php-bz2 php-gd php-sqlite3 php-xml && \
+	apt-get install -y --no-install-recommends cmake cmake-data mesa-utils vulkan-tools unzip apt-file wget && \
```
Comment (GabyCT) [inline tests/metrics/machine_learning/onednn-dockerfile/Dockerfile:21]: Do we need `wget`? seems like we are not using it

File: tests/metrics/machine_learning/openvino-dockerfile/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+
+RUN apt-get update && \
+	apt-get install -y --no-install-recommends git curl build-essential autoconf && \
+	apt-get install -y --no-install-recommends php libapache2-mod-php php-bz2 php-gd php-sqlite3 php-xml && \
+	apt-get install -y --no-install-recommends cmake cmake-data mesa-utils vulkan-tools unzip apt-file wget nano && \
+        curl -OkL ${PHORONIX_SRC} && \
+	tar zxf ${PHORONIX_TAR_FILE} && cd phoronix-test-suite && ./install-sh && \
+	apt-get clean && rm -rf /var/lib/apt/lists && \
+	rm -rf /var/lib/apt/lists/* && \
```
Comment (GabyCT) [inline tests/metrics/machine_learning/openvino-dockerfile/Dockerfile:28]: The same comments about `wget` and this step

File: tests/metrics/machine_learning/openvino.sh [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+	info "Restoring kata configuration"
+	[ -z "${KATA_CONFIG_FNAME}" ] && return
+
+	pushd "${KATA_DIR}"
+		sudo ln -sf "${KATA_CONFIG_FNAME}" "${SRC_CONFIG}"
+	popd
+}
+trap restore_kata_config EXIT
+
+clean_cache() {
```
Comment (GabyCT) [inline tests/metrics/machine_learning/openvino.sh:48]: what about making this function as part of the common of metrics? this is being used in onednn and openvino ml benchmarks

**Comment by GabyCT** [comment]
@dborquez thanks for the PR, overall lgtm, just a few comments

File: tests/metrics/machine_learning/onednn-dockerfile/Dockerfile (hunk 1 above)
Comment (dborquez) [reply tests/metrics/machine_learning/onednn-dockerfile/Dockerfile]: Fixed. Thank you @GabyCT

File: tests/metrics/machine_learning/onednn.sh [hunk 6]
```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+#
+# Copyright (c) 2020-2024 Intel Corporation
```
[the remark this answers is not in the document]
Comment (dborquez) [reply tests/metrics/machine_learning/onednn.sh]: Thank you @GabyCT, fixed!

File: tests/metrics/machine_learning/onednn.sh (hunk 2 above)
Comment (dborquez) [reply tests/metrics/machine_learning/onednn.sh]: good catch, fixed. Thank you @GabyCT

File: tests/metrics/machine_learning/openvino.sh [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+TEST_RESULTS_FNAME="${TMP_DIR}/openvino-results.json"
+
+readonly KATA_DIR="/opt/kata/share/defaults/kata-containers"
+readonly SRC_CONFIG="configuration.toml"
+
+# Variable used to store the initial configuration file name.
+# This file is again pointed to by kata once the script finishes.
+KATA_CONFIG_FNAME=""
+
+function restore_kata_config() {
```
[the remark this answers is not in the document]
Comment (dborquez) [reply tests/metrics/machine_learning/openvino.sh:37]: Good recomendation, changes applied. Thank you @GabyCT

File: tests/metrics/machine_learning/openvino.sh (hunk 5 above)
Comment (dborquez) [reply tests/metrics/machine_learning/openvino.sh]: done, thank you @GabyCT

File: tests/metrics/lib/common.bash [hunk 8]
```diff
@@ -425,3 +440,75 @@ function check_containers_are_running() {
...
+        pushd "${DEFAULT_KATA_CONFIG_DIR}" > /dev/null
+
+                if [ ! -f "${DEFAULT_KATA_CONFIG_FNAME}" ]; then
+                        die "Kata config file not found."
+                fi
+
+                info "Changing the kata configuration to assign '${NUM_CPUS} vcpus' and ${MEM_AVAIL_MB} MB of memory to the performance workload."
+
+                cp "${DEFAULT_KATA_CONFIG_FNAME}" "${WORKLOAD_CONFIG_FILE}"
+                ln -sf "${WORKLOAD_CONFIG_FILE}" "${DEFAULT_KATA_CONFIG_FNAME}"
```
[the remark this answers is not in the document]
Comment (dborquez) [reply tests/metrics/lib/common.bash:472]: @cmaf the copy of config pointed to by `WORKLOAD_CONFIG_FILE` is automatically deleted  when the scripts ends.

**Review by cmaf — APPROVED** [review]

**Pull request merged**